mmselfsup/models/backbones/mae_cls_vit.py

```diff
@@ -0,0 +1,116 @@
+            `with_cls_token` must be True. Defaults to True.
+        interpolate_mode (str): Select the interpolate mode for position
+            embeding vector resize. Defaults to "bicubic".
+        patch_cfg (dict): Configs of patch embeding. Defaults to an empty dict.
+        layer_cfgs (Sequence | dict): Configs of each transformer layer in
+            encoder. Defaults to an empty dict.
+        global_pool (bool): Whether or not use the global_pooling feature
+            for classification. Defaults to True.
+        finetune (bool): Whether or not do fine-tuning. Defaults to True.
+        init_cfg (dict, optional): Initialization config dict.
+            Defaults to None.
+    """
+
+    def __init__(self,
+                 arch='b',
+                 img_size=224,
+                 patch_size=16,
+                 out_indices=-1,
+                 drop_rate=0,
+                 drop_path_rate=0,
+                 norm_cfg=dict(type='LN', eps=1e-6),
+                 final_norm=True,
+                 output_cls_token=True,
+                 interpolate_mode='bicubic',
+                 patch_cfg=dict(),
+                 layer_cfgs=dict(),
+                 global_pool=True,
+                 finetune=True,
+                 init_cfg=None):
+        super().__init__(arch, img_size, patch_size, out_indices, drop_rate,
+                         drop_path_rate, norm_cfg, final_norm,
+                         output_cls_token, interpolate_mode, patch_cfg,
+                         layer_cfgs, init_cfg)
+
+        self.embed_dims = self.arch_settings['embed_dims']
+        self.global_pool = global_pool
+        if self.global_pool:
+            _, self.fc_norm = build_norm_layer(
+                norm_cfg, self.embed_dims, postfix=1)
+
+        self.finetune = finetune
+        if not self.finetune:
+            self._freeze_stages()
+
+    def _init_weights(self, m):
+        if isinstance(m, nn.Linear):
+            nn.init.xavier_uniform_(m.weight)
+            if isinstance(m, nn.Linear) and m.bias is not None:
+                nn.init.constant_(m.bias, 0)
+        elif isinstance(m, nn.LayerNorm):
+            nn.init.constant_(m.bias, 0)
+            nn.init.constant_(m.weight, 1.0)
+
+    def train(self, mode=True):
+        super(MAEClsViT, self).train(mode)
+        if not self.finetune:
+            self._freeze_stages()
+
+    def _freeze_stages(self):
+        for _, param in self.named_parameters():
+            param.requires_grad = False
+
+    def forward(self, x):
+        B = x.shape[0]
+        x = self.patch_embed(x)
+
+        # stole cls_tokens impl from Phil Wang, thanks
+        cls_tokens = self.cls_token.expand(B, -1, -1)
+        x = torch.cat((cls_tokens, x), dim=1)
+        x = x + self.pos_embed
+        x = self.drop_after_pos(x)
+
+        for i, layer in enumerate(self.layers):
+            x = layer(x)
+
+            if i == len(self.layers) - 1 and self.final_norm:
```

Comment:
1. line 106-114 is confusing, is `self.final_norm=True and self.global_pool=True` expected? It seems that `self.final_norm=True and self.global_pool=False` will go to line 113.
2. Considering move global_pool to neck so that we can reuse backbone in mmcls.